driver/config/test_config.go

```diff
@@ -51,25 +51,14 @@ func WithConfigValue(ctx context.Context, key string, value any) context.Context
 	return WithConfigValues(ctx, map[string]any{key: value})
 }
 
-func WithConfigValues(ctx context.Context, newValues map[string]any) context.Context {
-	values, ok := ctx.Value(contextConfigKey).(mapProvider)
+func WithConfigValues(ctx context.Context, setValues map[string]any) context.Context {
+	values, ok := ctx.Value(contextConfigKey).([]map[string]any)
 	if !ok {
-		values = make(mapProvider)
-	}
-	expandedValues := make([]map[string]any, 0, len(newValues))
-	for k, v := range newValues {
-		parts := strings.Split(k, ".")
-		val := map[string]any{parts[len(parts)-1]: v}
-		if len(parts) > 1 {
-			for i := len(parts) - 2; i >= 0; i-- {
-				val = map[string]any{parts[i]: val}
-			}
-		}
-		expandedValues = append(expandedValues, val)
-	}
-	for _, v := range expandedValues {
-		maps.Merge(v, values)
+		values = make([]map[string]any, 0)
 	}
+	newValues := make([]map[string]any, len(values), len(values)+1)
+	copy(newValues, values)
+	newValues = append(newValues, setValues)
 
-	return context.WithValue(ctx, contextConfigKey, values)
+	return context.WithValue(ctx, contextConfigKey, newValues)
```

Comment:
Kinda copied from your branch, but adjusted slightly to be a slice of maps. The benefit of that is that overrides still work in the correct order deterministically:
```go
ctx = WithConfigValues(ctx, map[string]any{"serve": map[string]any{"public": x}})
ctx = WithConfigValues(ctx, map[string]any{"serve.public": x})
```
The keys in this case would not be the same, so both would be part of the map after merge, but only one of the values would be used at random.